### PR TITLE
Rchain 3072: refactor SystemProcesses

### PR DESCRIPTION
**Pull request by odeac**
## Overview
cleanup in the SystemProcesses area to remove duplication as preparation for rchain-3072

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3072

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/RhoType.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      }
+  }
+
+
+  object Number {
+    def unapply(p: Par): Option[Long] =
+      p.singleExpr().collect {
+        case Expr(GInt(v)) => v
+      }
+  }
```
Comment (KentShikama) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/RhoType.scala]: It always gets me how GInt is a Long.

File: project/Rholang.scala [hunk 2]
```diff
@@ -9,34 +9,6 @@ import sbt.io.Path.relativeTo
  */
 object Rholang {
```
Comment (tymm) [inline project/Rholang.scala]: Can we delete the whole file? IntelliJ show no usage for `lazy val rholangSettings`.

Comment (tymm) [reply project/Rholang.scala]: Nevermind, we use it in `build.sbt`. IntelliJ fooled me.

Comment (tymm) [reply project/Rholang.scala]: But what about moving it to `build.sbt` and deleting this file?

**Pull request merged**